### ISSUE TITLE
Not able to train toy experiment. [cuda runtime error (59) : device-side assert triggered at /pytorch]

### ISSUE THREAD

**Issue by naveennaidu** [MIC-DKFZ/medicaldetectiontoolkit#27]
CUDA: 9.0

I was following the steps mentioned in readme file. After running the code to train the toy experiment.
`CUDA_LAUNCH_BLOCKING=1 python exec.py --mode train --exp_source experiments/toy_exp/ --exp_dir exp_dir_toy/`

I got this error
![image](https://user-images.githubusercontent.com/28999143/54474726-5851ab00-480e-11e9-8784-e30e488d31df.png)

I think class labels are 1, 2 instead of 0, 1.
How can i solve this problem? Did anyone experienced this error?

**Comment by pfjaeger** [comment]
hi class labels 1,2 are ok, because 0 is for background. The error you encounter indicates that the range of labels (in this case it should be [0, 1, 2]) does not match the shape of your logits in the class dimension (which should be 3). Can you please print the range of labels as well as the logit dimensions to check which one is faulty? The error might be caused by a wrong setting in your configs. you could print your configs as well so i could check.

**Comment by naveennaidu** [comment]
Hey, Thanks for replying.
My error is coming at [line 447](https://github.com/pfjaeger/medicaldetectiontoolkit/blob/2b343ba974a6936d10bc7ba2d571db28f0760df5/models/retina_unet.py#L447) which is cross entropy. 
I checked input and target shape and got **input->[4, 2, 320, 320]** , **target-> [4, 320, 320]**
But input shape should be [4, 3, 320, 320].
I printed num_seg_classes and got 2. Now I know what the error is. 

I'm writing this for anyone who faces the same issue as me. Check num_seg_classes(print at [line 378](https://github.com/pfjaeger/medicaldetectiontoolkit/blob/2b343ba974a6936d10bc7ba2d571db28f0760df5/models/retina_unet.py#L378) and know the value of **_self.cf.num_seg_classes_**)
